rxandroidble/src/main/java/com/polidea/rxandroidble/internal/RxBleDeviceImpl.java

```diff
@@ -42,17 +44,17 @@ public RxBleDeviceImpl(BluetoothDevice bluetoothDevice, RxBleConnection.Connecto
     public Observable<RxBleConnection> establishConnection(Context context, boolean autoConnect) {
         return Observable.defer(() -> {
 
-                if (isConnected.compareAndSet(false, true)) {
-                    return connector.prepareConnection(context, autoConnect)
-                            .doOnSubscribe(() -> connectionStateSubject.onNext(CONNECTING))
-                            .doOnNext(rxBleConnection -> connectionStateSubject.onNext(CONNECTED))
-                            .doOnUnsubscribe(() -> {
-                                connectionStateSubject.onNext(DISCONNECTED);
-                                isConnected.set(false);
-                            });
-                } else {
-                    return Observable.error(new BleAlreadyConnectedException(bluetoothDevice.getAddress()));
-                }
+            if (isConnected.compareAndSet(false, true)) {
```

Comment:
I don't see where `isConnected` is set to `true` - it seems to be always `false`

Reply:
At this very line it is set to `true` if the current state equals `false`